OpenRA.Mods.Common/Widgets/Logic/Ingame/IngameMenuLogic.cs

```diff
@@ -88,14 +88,20 @@ public IngameMenuLogic(Widget widget, World world, Action onExit, WorldRenderer
 
 			abortMissionButton.OnClick = () =>
 			{
-				if (world.IsGameOver)
-				{
-					onQuit();
-					return;
-				}
-
 				hideMenu = true;
-				ConfirmationDialogs.PromptConfirmAction("Abort Mission", "Leave this game and return to the menu?", onQuit, showMenu);
+				var restartText = "";
+				if (world.IsReplay || world.Map.Visibility == MapVisibility.MissionSelector)
+					restartText = "Restart";
```

Comment:
Once you've extracted the restart-specific code out of the dialog code and toggle the button visibility based on the action, you can then set the restartAction to be non-null here (basically just assign the block that you pull out of the helper).